js/thread-client.js

```diff
@@ -0,0 +1,34 @@
+const { DebuggerClient } = require('devtools/shared/client/main');
+const { DebuggerTransport } = require('devtools/transport/transport');
+const { TargetFactory } = require("devtools/client/framework/target");
+const promise = require('devtools/sham/promise');
+
+const socket = new WebSocket("ws://localhost:9000");
+const transport = new DebuggerTransport(socket);
+const client = new DebuggerClient(transport);
+
+function connectToClient(onConnect) {
+  client.connect().then(() => {
+    return client.listTabs().then(onConnect);
+  }).catch(err => console.log(err));
+}
+
+function connectToTab(tab, onNewSource) {
+  let deferred =  promise.defer();
+  const options = { client, form: tab, chrome: false };
+
+  TargetFactory.forRemoteTab(options).then(target => {
+    target.activeTab.attachThread({}, (res, threadClient) => {
+      threadClient.resume();
+      window.gThreadClient = threadClient;
```

Comment:
I tried three different hacks for exporting `gThreadClient`, which would let us do something like:

```js
const {gThreadClient} = require('../thread-client');

function loadSources() {
  gThreadClient.getSources();
}
```

1.  exporting a reference to the threadClient
  + `module.exports.gThreadClient = threadClient`  
  + webpack would get a reference to the module before the exports property had been set
2. exporting a stub that received threadClient properties
  + `Object.assign(gThreadClient, threadClient, threadClient.__proto__)`
  + webpack would get a reference to an object with all of the properties except the threadClient's prototype getters, which we could get if we did a slightly better deep clone than Object.assign
3. exporting a getter
  + `module.exports = {get gThreadClient () {return threadClient}}`
  + webpack would get a reference to threadClient before it was set.

Ultimately, I decided to leave `gThreadClient` a global for the time being. :)